src/main/java/uk/ac/cam/cl/dtg/isaac/dos/eventbookings/PgEventBookings.java

```diff
@@ -114,71 +114,185 @@ public EventBooking add(final String eventId, final Long userId, final Long rese
     }
 
     @Override
-    public EventBooking add(final String eventId, final Long userId, final BookingStatus status, Map<String, String> additionalEventInformation) throws SegueDatabaseException {
-        return add(eventId, userId, null, status, additionalEventInformation);
+    public EventBooking add(final ITransaction transaction, final String eventId, final Long userId, final BookingStatus status,
+                            final Map<String, String> additionalEventInformation) throws SegueDatabaseException {
+        return add(transaction, eventId, userId, null, status, additionalEventInformation);
     }
 
     @Override
-    public void updateStatus(final String eventId, final Long userId, final Long reservingUserId, final BookingStatus status, final Map<String, String> additionalEventInformation) throws SegueDatabaseException {
-        PreparedStatement pst;
-        // FIXME: try-with-resources!
-        try (Connection conn = ds.getDatabaseConnection()) {
-
-            String reservingUserIdClause = "";
-            if (reservingUserId != null) {
-                reservingUserIdClause = ", reserved_by = ? ";
-            }
+    public void updateStatus(final ITransaction transaction, final String eventId, final Long userId, final Long reservingUserId, final BookingStatus status, final Map<String, String> additionalEventInformation) throws SegueDatabaseException {
```

Comment:
You can use `target/site/jacoco/index.html` to see the test coverage reports to see how much of the code has coverage. Unfortunately this only applies to the unit tests and not the end-to-end tests, and we don't test the database layer for unit tests because I don't think it would be meaningful.

I am happy to assert that this is safe to deploy if we tell the teams to keep an eye on it!